mmcv/runner/hooks/hook.py

```diff
@@ -65,3 +69,28 @@ def is_last_epoch(self, runner):
 
     def is_last_iter(self, runner):
         return runner.iter + 1 == runner._max_iters
+
+    def get_trigger_stages(self):
+        trigger_stages = set()
+        for stage in Hook.stages:
+            base_method = getattr(Hook, stage)
```

Comment:
Mak create a wrapper named `check_stage_implemented(hook, stage)` to reduce lines 76-78 & 91-93